src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClonesArray = [];
+  const stateCopyObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopyObject, action.extraData);
+        stateClonesArray.push({ ...stateCopyObject });
```

Comment:
The state clone is pushed into the array after each operation. This could be optimized by pushing the state clone into the array only once, after all operations in the switch statement. This can be achieved by moving this line after the switch statement.